### PR TITLE
Soru+hs/tslint

### PR DESCRIPTION
**Pull request by Sorunome**
Continuuing of #92

### CODE REVIEW

**Comment by Half-Shot** [comment]
>soru isn't sure what the purpose of no-empty is, as empty objects can make sense indeed

It's for if / loop/ function blocks with no contents .e.g
```javascript
if (thing) {

}
```

>IMO max-file-line-count is kinda meh, it triggers on two files currently

Yeah, it's just annoying that we have massive files but both of them could be broken up a bit.

>we should do something about no-any, as that warns on like all the time an event is used, for example.

We really ought to define some interfaces for events, to fix this. Ideally the bridge lib would provide them but hey ho.

**Comment by Sorunome** [comment]
> It's for if / loop/ function blocks with no contents .e.g

it also triggers on empty objects - what do?

> Yeah, it's just annoying that we have massive files but both of them could be broken up a bit.

those sound like different PRs? or should we do it in here somehow?

**Comment by Half-Shot** [comment]
Do we use object ``{}`` extensively, and we can't replace them with something else?

Also, I'd stick the max line thing on warn.

**Comment by Half-Shot** [comment]
While I'm at it, additional rules to consider:

- https://palantir.github.io/tslint/rules/promise-function-async/
- https://palantir.github.io/tslint/rules/await-promise/
- https://palantir.github.io/tslint/rules/no-bitwise/
- https://palantir.github.io/tslint/rules/no-debugger/
- https://palantir.github.io/tslint/rules/no-floating-promises/
- https://palantir.github.io/tslint/rules/no-unused-expression/
- https://palantir.github.io/tslint/rules/prefer-const/

**Comment by Sorunome** [comment]
> Do we use object `{}` extensively, and we can't replace them with something else?

Tests, tools, and one code place, it seems

**Comment by Half-Shot** [comment]
Oh good lord, tests...

**Comment by Half-Shot** [comment]
> > Do we use object `{}` extensively, and we can't replace them with something else?
> 
> Tests, tools, and one code place, it seems

I guess we could use `Object()`, or just drop the rule for now :)

File: src/matrixeventprocessor.ts [hunk 1]
```diff
@@ -249,7 +258,7 @@ export class MatrixEventProcessor {
         return [embed, reponseText];
     }
 
-    private async SetEmbedAuthor(embed: Discord.RichEmbed, sender: string, profile?: any) {
+    private async SetEmbedAuthor(embed: Discord.RichEmbed, sender: string, profile?: IMatrixEvent | null) {
```
[the remark this answers is not in the document]
Comment (Sorunome) [reply src/matrixeventprocessor.ts]: It seems like basically everything from matrix are events?

[the remark this answers is not in the document]
Comment (Half-Shot) [reply src/matrixeventprocessor.ts]: Not this one, it's a ``{displayname ,avatar_url}`` only. The profiles we get from the request don't contain any event keys. I'd make a separate interface.

**Comment by Half-Shot** [comment]
Oh, we can also drop ``--type-check`` from https://github.com/Half-Shot/matrix-appservice-discord/pull/270/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R8 now apparently.

File: src/db/postgres.ts [hunk 2]
```diff
@@ -1,10 +1,9 @@
-import {IMain, IDatabase} from "pg-promise";
```
[the remark this answers is not in the document]
Comment (Sorunome) [reply src/db/postgres.ts]: else duplicate import with `import * as pgPromise from "pg-promise";`

**Comment by Sorunome** [comment]
since discord.js typings updated since this and soru wanted that this builds it also tackles #285 now

**Pull request merged**